src/projects/extension/script/commands/check-syntax.ts

```diff
@@ -0,0 +1,45 @@
+import { ConnectionProvider } from "@core/public.api";
+import { QixApplicationProvider } from "@core/qix/utils/application.provider";
+import { EntryType } from "@vsqlik/fs/data";
+import { container } from "tsyringe";
+import * as vscode from "vscode";
+
+const appProvider           = container.resolve(QixApplicationProvider);
+const connectionProviver    = container.resolve(ConnectionProvider);
+const diagnosticsCollection = vscode.languages.createDiagnosticCollection("vsqlik.script_errors");
+
+/**
+ * checks the script syntax for a qlik script
+ */
+export async function CheckScriptSyntax(uri: vscode.Uri): Promise<void> {
+
+    /** get required data, connection, app and the text editor which contains the script */
+    const connection = await connectionProviver.resolve(uri);
+    const appEntry   = connection?.fileSystem.parent(uri, EntryType.APPLICATION);
+    const textEditor = vscode.window.visibleTextEditors.find(
+        (editor) => editor.document.uri.toString(true) === uri.toString(true));
+
+    if (!connection || !appEntry || !textEditor) {
```

Comment:
please create a virtual textEditor if the current editor is not visible to fetch the positionAt

Reply:
I changed to logic a bit and dont use the visible text editors anymore, since i know the textpos i simply take the substr from 0 to error position and split this by EOL, so then i know the line number.